packages/website/plugins/blog-footer.ts

```diff
@@ -0,0 +1,52 @@
+import type * as mdast from 'mdast';
+import type { Plugin } from 'unified';
+
+import { nodeIsParent } from './utils/nodes';
+
+export const blogFooter: Plugin = () => {
+  return (root, file) => {
+    if (
+      !nodeIsParent(root) ||
+      !(file.value as string).includes('<!--truncate-->')
```

Comment:
Agh, noticed this morning - this plugin is also active in the root blog listing (https://deploy-preview-9238--typescript-eslint.netlify.app/blog):

![Screenshot of the root blog list, showing the 'Supporting typescript-eslint' h2 at the bottom of the first blog preview](https://github.com/typescript-eslint/typescript-eslint/assets/3335181/9c780066-f4e2-4340-a8f6-ca30e064b3b6)

@Josh-Cena is there a better way to stop that? I couldn't find anything searching around.

Reply:
I don't think a better way exists, unfortunately. For context, this is the way it works:

- When Docusaurus renders the overview page, it uses the same syntax to import Markdown, but adds an extra query: `require("some-post.mdx?truncated=true")`
- The blog plugin creates two Webpack loaders, one that reads the resource query, and truncates the source if the `truncated` query exists; one that actually does the MDX transformation, etc.
- Therefore, the MDX loader that runs the plugins doesn't know the protocol of the `truncated` query at all.

First-class truncation is in https://github.com/facebook/docusaurus/issues/6480. We could make plugins know that a file is truncated by giving it the truncation query, for example.